lib/index.ts

```diff
@@ -0,0 +1,598 @@
+/*
+Copyright 2017 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as request from 'request-promise';
+
+const URL = 'https://api2.frontapp.com';
+
+interface Request {
+	reqType: string;
+	path: string;
+}
+
+type InternalCallback = (err: Error | null, response: any | null) => void;
+
+export class Front {
+	public comment: {
```

Comment:
I think you can actually use
```typescript
	public comment = {
		create: _.partial(this.httpCall,
			{ reqType: 'POST', path: 'conversations/{conversation_id}/comments' }),
		get: _.partial(this.httpCall,
			{ reqType: 'GET', path: 'comments/{comment_id}' }),
		listMentions: _.partial(this.httpCall,
			{ reqType: 'GET', path: 'comments/{comment_id}/mentions' }),
	};
```
here and it will still have the same code output as the current one (where it gets initialised in the constructor, but without having to manually declare all of the types here (at least I think they will automatically come for free?)